File: Algo.js
const breadthFirstSearch = (graph, startNode, endNode) => {
    const queue = [startNode];
    const visited = [];
    while (queue.length) {
        const currentNode = queue.shift();
        if (currentNode === endNode) {
        return visited;
        }
        if (!visited.includes(currentNode)) {
        visited.push(currentNode);
        queue.push(...graph[currentNode]);
        }
    }
    return visited;
    }

File: Algo.py
def breadthFirstSearch(graph, startNode, endNode):
    queue = [startNode]
    visited = []
    while queue:
        currentNode = queue.pop(0)
        if currentNode == endNode:
            return visited
        if currentNode not in visited:
            visited.append(currentNode)
            queue.extend(graph[currentNode])
    return visited